api/v1alpha1/limitador_types.go

```diff
@@ -285,6 +332,20 @@ func (s *LimitadorStatus) Equals(other *LimitadorStatus, logger logr.Logger) boo
 	return true
 }
 
+type Image struct {
+	// Repository containing the image to be used
+	// +optional
+	Repository *string `json:"repository"`
+
+	// Tag of the image
+	// +optional
+	Tag *string `json:"tag"`
+
+	// PullSecret for pulling private images
+	// +optional
+	PullSecret *corev1.LocalObjectReference `json:"pullSecret"`
```

Comment:
This is interesting. Not implemented in other CRD controllers. I would have

```go
type LimitadorSpec struct {
    // +optional
    Image *string `json:"image,omitempty"` 

    // +optional
    ImagePullSecret *corev1.LocalObjectReference `json:"imagePullSecret,omitempty"`
```

wdyt?

Reply:
Yup, this looks good to me 👍